### DISCUSSION TITLE
Optimization stops at first iteration

### DISCUSSION THREAD

**Discussion by msenzacqua** [anyoptimization/pymoo#208] [Q&A] [answered]
Hello,

for our optimization problem we set the termination criterion at 30 generations. 
But during the optimization run number of function evaluations is stuck at 40 and the optimization stops at the second generation.
What could be the issue?

<img width="680" alt="Schermata 2021-11-10 alle 11 46 47" src="https://user-images.githubusercontent.com/46441068/141100007-6c0a52b0-84d2-4cb2-88e6-8124256b8f3b.png">

Thank you for the help

Martina

**Comment by blankjul** [comment]
Is your optimization problem very small? It seems like in the second generation no new solutions can be found. This is indicated by `n_eval` being 40 in the first AND second generations. What algorithm are you running? What is your search space?

**Reply by msenzacqua** [reply]
The problem is actually small (only two elements to estimate for the solution array).  I am running the NSGA2 algorithm and the search space has two dimensions

**Comment by blankjul** [comment]
Is it real-valued or discrete? If discrete, how big is the search space?

**Reply by msenzacqua** [reply]
it is discrete-valued and the search space amplitude is 5 for each variable

**Comment by blankjul** [comment, marked as answer]
Which means your search space is 5*5 = 25 right? 
The algorithm can not find new non-duplicate solutions then. This is why it terminates.

**Reply by msenzacqua** [reply]
Ok, now it is clear. Thank you very much Julian!
